### ISSUE TITLE
Support NVidia CUDA compilation with nvcc

### ISSUE THREAD

**Issue by fhunleth** [nerves-project/toolchains#56]
Pelemay uses `nvcc` to generate CUDA code and would require it when targeting devices with NVidia GPUs (like the Jetson Nano)

**Comment by zacky1972** [comment]
Thank you.

Usually, `nvcc` is located at `/usr/local/cuda/bin` at least in case of Ubuntu x86_64.

**Comment by mobileoverlord** [comment]
Is this an option that could be brought into the tool chain via CTNG?

**Comment by fhunleth** [comment]
I'm quite sure that the answer is no. Given that it's NVidia, I bet that it's a set of precompiled binaries and header files that need to be plucked out of their SDK. It would be way better to have a separate "toolchain" archive for it, but I don't know how to do that.

**Comment by zacky1972** [comment]
I mistook it.

To cross-compile `cu` code by `nvcc`, we set `-ccbin` and `-Xcompiler` options to a cross-compiler of C++, and its compiling options, respectively.

Does Nerves set CXX environment variable?

**Comment by fhunleth** [comment]
Yes, `CXX` is set and the compiler options are in `CXXFLAGS`. Here's where it's set: https://github.com/nerves-project/nerves_system_br/blob/master/nerves_env.exs#L124-L136.

**Comment by zacky1972** [comment]
That's great! Thus, I can implement Pelemay to work Jetson, even if on Nerves in future.
